src/Compilers/CSharp/Portable/Symbols/Synthesized/Records/SynthesizedRecordEquals.cs

```diff
@@ -155,7 +155,7 @@ internal override void GenerateMethodBody(TypeCompilationState compilationState,
                 }
 
                 fields.Free();
-
+                retExpr = F.LogicalOr(F.ObjectEqual(F.This(), other), retExpr);
```

Comment:
>F.LogicalOr [](start = 26, length = 11)

I think we should do similar change for the other consumer of the ```GenerateFieldEquals``` helper, the code that generates Equals for Anonymous Types. I also think that we should check/adjust code gen for VB Anonymous Types as well.

Reply:
@AlekseyTs That sounds reasonable. Should I start doing this and update tests? or do you want to wait for another team member to confirm?

Reply:
>> I think we should do similar change for the other consumer of the ```GenerateFieldEquals``` helper, the code that generates Equals for Anonymous Types. I also think that we should check/adjust code gen for VB Anonymous Types as well.

> That sounds reasonable. Should I start doing this and update tests? or do you want to wait for another team member to confirm?

I don't think we need to wait for the confirmation from other team members. If I was working on this change, I would just do this (the improvement of code gen for Anonymous Types). 

---
In reply to: [575327671](https://github.com/dotnet/roslyn/pull/51149#discussion_r575327671) [](ancestors = 575327671)

Reply:
@AlekseyTs I made the change for anonymous types for C# and updated the tests.

Could you point me to where the parallel change for VB should go?

I looked at https://github.com/Youssef1313/roslyn/blob/records-ref-equality/src/Compilers/VisualBasic/Portable/Symbols/AnonymousTypes/SynthesizedSymbols/AnonymousType_EqualsMethodSymbol.vb but didn't find anything relevant. So I'm probably in the wrong place.

Reply:
> Could you point me to where the parallel change for VB should go?

See ```AnonymousType_IEquatable_EqualsMethodSymbol.GetBoundMethodBody``` in src\Compilers\VisualBasic\Portable\Binding\SyntheticBoundTrees\AnonymousTypeSyntheticMethods.vb.

---
In reply to: [575376850](https://github.com/dotnet/roslyn/pull/51149#discussion_r575376850) [](ancestors = 575376850)